src/home/homectl.c

```diff
@@ -4511,6 +4518,56 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_TMP_LIMIT:
+                case ARG_DEV_SHM_LIMIT: {
+                        const char *field =
+                                    c == ARG_TMP_LIMIT ? "tmpLimit" :
+                                c == ARG_DEV_SHM_LIMIT ? "devShmLimit" : NULL;
+                        const char *field_scale =
+                                    c == ARG_TMP_LIMIT ? "tmpLimitScale" :
+                                c == ARG_DEV_SHM_LIMIT ? "devShmLimitScale" : NULL;
+
+                        assert(field);
+                        assert(field_scale);
+
+                        if (isempty(optarg)) {
+                                r = drop_from_identity(field);
+                                if (r < 0)
+                                        return r;
+                                r = drop_from_identity(field_scale);
+                                if (r < 0)
+                                        return r;
+                                break;
+                        }
+
+                        r = parse_permyriad(optarg);
+                        if (r < 0) {
+                                uint64_t u;
+
+                                r = parse_size(optarg, 1024, &u);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse %s parameter: %s", field, optarg);
+
+                                r = sd_json_variant_set_field_unsigned(&arg_identity_extra, field, u);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to set %s field: %m", field);
+
+                                r = drop_from_identity(field_scale);
```

Comment:
userdb itself supports setting both absolute and scaled value. Why restricted to one of them in homectl?

Reply:
well, the userdb can lots of things we currently don't expose in homectl, such as configurable per-machine fields. 

the primary reason why those things aren't exposed is because it's not obvious how to do this nicely from the cmdline. It's certainly doable, but so far I didn't want to spend time on designing this. i mean, we'd probably have to come up with `--tmp-quota=5%:17M` or so, but it wasn't clear to me that this is really the way to go...

hence i cut it short and just allow absolute or relative but not both.

but that just means: for now, if someone has a good reason to set both later, we can look into that.